[Resubmit] state_dict_pre_hook

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #90435

Resubmit of https://github.com/pytorch/pytorch/pull/88541 which got stale.

rohan-varma committed:
[Resubmit] state_dict_pre_hook

[ghstack-poisoned]

fegin reviewed (approved):
Thanks for adding this hook! Stamp it as it is a resubmit one.

rohan-varma committed:
Update on "[Resubmit] state_dict_pre_hook"


Resubmit of https://github.com/pytorch/pytorch/pull/88541 which got stale.


[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge -f

rohan-varma commented:
@pytorchbot merge
